lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -0,0 +1,949 @@
+#include "TypeChecker.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/Expr.h"
+#include "swift/AST/Module.h"
+#include "swift/AST/ParameterList.h"
+#include "swift/AST/Pattern.h"
+#include "swift/AST/Stmt.h"
+#include "swift/AST/Types.h"
+#include "DerivedConformances.h"
+
+using namespace swift;
+using namespace DerivedConformance;
+
+// Returns whether the type represented by the given ClassDecl inherits from a
+// type which is Codable.
+static bool classInheritsCodable(TypeChecker &tc, ClassDecl *type) {
+  if (!type || !type->hasSuperclass())
```

Comment:
Yes, that sounds like a better name. Also, can you remove the '!type' check or replace it with an assertion? It's not clear to me that this can happen and we should not let it happen silently if it does.